### PR TITLE
Babi memory network 95% in 65 epochs (+2%)

### PR DESCRIPTION
**Pull request by farizrahman4u**

### CODE REVIEW

File: examples/babi_memnn.py [hunk 1]
```diff
@@ -184,10 +185,11 @@ def vectorize_stories(data, word_idx, story_maxlen, query_maxlen):
 answer = Sequential()
 answer.add(Merge([response, question_encoder], mode='concat', concat_axis=-1))
 # the original paper uses a matrix multiplication for this reduction step.
-# we choose to use a RNN instead.
-answer.add(LSTM(64))
-# one regularization layer -- more would probably be needed.
-answer.add(Dropout(0.25))
+# we choose to use a RNN stack instead.
+answer.add(LSTM(30, return_sequences=True))
```
Comment (fchollet) [inline examples/babi_memnn.py]: The problem with this script is 100% overfitting, not underfitting. Adding more layers actually increase overfitting (as well as increasing training time). 

Your gains are due to the higher dropout value, plus the fact that you are using less LSTM parameters (which also reduces overfitting). But you can get even bigger gains just by adding Dropout layers, which come almost for free : )

I updated the example with a version that gets to 98.6% accuracy and runs faster than the previous script.

**Comment by jeffzhengye** [comment]
It seems there is a softmax for match in the paper.

**Comment by farizrahman4u** [comment]
@fchollet Cool.. so the gain was from the dropout. I thought it was because of the 'narrow and deep better than wide and shallow' thing.

**Comment by fchollet** [comment]
As long as you are not overfitting, then deeper nets are likely to help. But if you are overfitting, the simplest thing to do is to increase regularization...
